azurerm/resource_arm_app_service_plan.go

```diff
@@ -0,0 +1,133 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/arm/web"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmAppServicePlan() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAppServicePlanCreateUpdate,
+		Read:   resourceArmAppServicePlanRead,
+		Update: resourceArmAppServicePlanCreateUpdate,
+		Delete: resourceArmAppServicePlanDelete,
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"location": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"tier": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
Looking at the `SkuDescription` object:

```
// SkuDescription is description of a SKU for a scalable resource.
type SkuDescription struct {
	Name         *string       `json:"name,omitempty"`
	Tier         *string       `json:"tier,omitempty"`
	Size         *string       `json:"size,omitempty"`
	Family       *string       `json:"family,omitempty"`
	Capacity     *int32        `json:"capacity,omitempty"`
	SkuCapacity  *SkuCapacity  `json:"skuCapacity,omitempty"`
	Locations    *[]string     `json:"locations,omitempty"`
	Capabilities *[]Capability `json:"capabilities,omitempty"`
}
```

I think it makes sense to separate this out into it's own block i.e.:

```
sku {
  name = "Standard_S0"
  tier = "Standard"
  size = "S0"
  ...
}
```

Here's an [example of the Schema usage](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_container_service.go#L48) - [expanding/parsing it out](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_container_service.go#L474) and [flattening/writing it back](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_container_service.go#L334) - which I'm hoping makes sense?